lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -192,17 +185,14 @@ def warn_about_bare_error!
                          "_positives = :nothing`")
         end
 
-        def warn_about_nil_error!
-          handle_warning("Using the `raise_error` matcher with a `nil` error is probably " \
-                         "unintentional, it risks false positives, since `raise_error` " \
-                         "will match when Ruby raises a `NoMethodError`, `NameError` or " \
-                         "`ArgumentError`, potentially allowing the expectation to pass " \
-                         "without even executing the method you are intending to call. " \
-                         "#{warning}"\
-                         "Instead consider providing a specific error class or message. " \
-                         "This message can be suppressed by setting: " \
-                         "`RSpec::Expectations.configuration.on_potential_false" \
-                         "_positives = :nothing`")
+        def raise_nil_argument_error!
+          raise "Using the `raise_error` matcher with a `nil` error is probably " \
+                "unintentional, it risks false positives, since `raise_error` " \
+                "will match when Ruby raises a `NoMethodError`, `NameError` or " \
+                "`ArgumentError`, potentially allowing the expectation to pass " \
+                "without even executing the method you are intending to call. " \
+                "#{warning}"\
+                "Instead consider providing a specific error class or message."
```

Comment:
```suggestion
          raise ArgumentError, "Using the `raise_error` matcher with a `nil` " \
                "value is probably unintentional, it risks false positives, since " \
                "`raise_error` will match when Ruby raises a `NoMethodError`" \
                ", `NameError` or `ArgumentError`, potentially allowing the " \
                "expectation to pass without even executing the method you "\
                "are intending to call. " \
                "#{warning}"\
                "Instead provide a specific error class or message."
```